packages/datetime/src/datePicker.tsx

```diff
@@ -246,7 +246,7 @@ export class DatePicker extends AbstractPureComponent<IDatePickerProps, IDatePic
                 selectedDay: day.getDate(),
             });
         }
-        if (this.state.value == null || this.state.value.getMonth() !== day.getMonth()) {
+        if (this.state.value !== null && this.state.value.getMonth() !== day.getMonth()) {
```

Comment:
`!= null` checks for null or undefined. please change.

Reply:
Will change to `!=` and add test @giladgray

Reply:
Changed to `!=` operator and added one test.